### ISSUE TITLE
expected type: Number, found: String, even though schema requires an Integer

### ISSUE THREAD

**Issue by bilalba** [everit-org/json-schema#381]
For the following schema
```
{
  "$id": "https://example.com/person.schema.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Person",
  "type": "object",
  "properties": {
    "firstName": {
      "type": "integer",
      "description": "The person's first name."
    }
  }
}
```

and this object:  `{"firstName":"1.123"}`

The error that I get is:

> expected type: Number, found: String

when it should really say:

> expected type: Integer, found: String

I'm working on type-casting the object values to the correct type defined in the schema, and when I see the above error, it's ambiguous whether I should type-cast it to an integer or a number.

**Comment by erosb** [comment]
Hello, this is now fixed by @zgyorffi in #387 . Thanks for reporting the issue.

**Comment by bilalba** [comment]
Thanks @zgyorffi @erosb
